packages/client/tests/functional/methods/findFirstOrThrow/tests.ts

```diff
@@ -42,7 +42,7 @@ testMatrix.setupTestSuite((suiteConfig, suiteMeta) => {
     expect(record).toBeNull()
   })
 
-  test('works with interactive transactions', async () => {
+  testIf(!process.env.DATA_PROXY)('works with interactive transactions', async () => {
```

Comment:
Yeah, I think it might, but it also might be a bit complicated.

It used to be that way before https://github.com/prisma/prisma/pull/13561, and was changed for reasons, as far as I remember. I feel like it did introduce some inconsistencies in the codebase with the need to have special case handling in some cases, like this: https://github.com/prisma/prisma/blob/d4e397ef893ce3e530adcbf5ea0d9e7bb1e101e2/packages/client/src/runtime/getPrismaClient.ts#L469
but at the same time, without looking at the diff in that PR, I believe it probably should have fixed some other inconsistencies.

The problem here is the ambiguity of the notion of an "engine type": there is _no_ engine of type "Data Proxy" as in the actual Query Engine flavor, but there _is_ a client-side engine abstraction of type `DataProxyEngine`, which is one of the three implementations of the `Engine` abstract class that we have. Wherever these two notions are conflated, there will be some inconsistencies in one or the other way. Thus these should be two different enums and the code might need some refactoring.

Let's discuss this together with @millsp, it will be a good topic for the Deep Dive meeting.

Reply:
(FWIW you are also commenting on an outdated diff here, right now it looks like `testIf(!clientMeta.dataProxy)`. It doesn't fundamentally change anything, but just pointing it out because judging from the other comment it looks like you have reviewed different files from different commits — perhaps something went wrong in GitHub UI)